File: scripts/fix_imports.py
import os
import re

def convert_to_package_import(file_path, base_dir, package_name):
    with open(file_path, 'r', encoding='utf-8') as f:
        content = f.read()

    # Regex to find relative imports: import '...'; or import "...";
    # We look for imports starting with . or ..
    import_pattern = re.compile(r"import\s+(['\"])(?P<path>\.\.?/.*?)\1")

    def replace_match(match):
        relative_path = match.group('path')
        quote = match.group(1)
        
        # Calculate the absolute path of the imported file
        file_dir = os.path.dirname(file_path)
        abs_imported_path = os.path.normpath(os.path.join(file_dir, relative_path))
        
        # Convert to package path
        # Expected lib structure: lib/...
        if 'lib' in abs_imported_path:
            # Find the part after 'lib'
            parts = abs_imported_path.split(os.sep)
            try:
                lib_index = parts.index('lib')
                package_relative = '/'.join(parts[lib_index + 1:])
                return f"import {quote}package:{package_name}/{package_relative}{quote}"
            except ValueError:
                return match.group(0) # Fallback if something is weird
        
        return match.group(0)

    new_content = import_pattern.sub(replace_match, content)

    if new_content != content:
        with open(file_path, 'w', encoding='utf-8') as f:
            f.write(new_content)
        return True
    return False

def main():
    base_dir = r"d:\Development\site_buddy\lib"
    package_name = "site_buddy"
    
    modified_count = 0
    for root, dirs, files in os.walk(base_dir):
        for file in files:
            if file.endswith('.dart'):
                file_path = os.path.join(root, file)
                if convert_to_package_import(file_path, base_dir, package_name):
                    modified_count += 1
                    print(f"Modified: {file_path}")

    print(f"\nTotal files modified: {modified_count}")

if __name__ == "__main__":
    main()

File: scripts/fix_imports.dart
import 'dart:io';

void main() {
  final libDir = Directory('lib');
  if (!libDir.existsSync()) return;

  int modifiedCount = 0;
  final entities = libDir.listSync(recursive: true);

  for (var entity in entities) {
    if (entity is File && entity.path.endsWith('.dart')) {
      if (processFile(entity)) {
        modifiedCount++;
        stdout.writeln('Modified: ${entity.path}');
      }
    }
  }
  stdout.writeln('\nTotal files modified: $modifiedCount');
}

bool processFile(File file) {
  final content = file.readAsStringSync();
  final lines = content.split('\n');
  bool changed = false;

  final newLines = lines.map((line) {
    var trimmed = line.trim();
    if ((trimmed.startsWith('import ') || trimmed.startsWith('export ')) &&
        trimmed.endsWith(';')) {
      // Find the path string between quotes
      String? path;
      String? quote;

      if (trimmed.contains("'")) {
        quote = "'";
      } else if (trimmed.contains('"')) {
        quote = '"';
      }

      if (quote != null) {
        final firstQuote = trimmed.indexOf(quote);
        final lastQuote = trimmed.lastIndexOf(quote);
        if (firstQuote != -1 && lastQuote != -1 && firstQuote != lastQuote) {
          path = trimmed.substring(firstQuote + 1, lastQuote);

          if (path.startsWith('./') || path.startsWith('../')) {
            // Resolve path
            final fileDir = file.parent.absolute.path;
            final absoluteImportPath = resolvePath(fileDir, path);

            final separator = Platform.pathSeparator;
            final libSearch = '${separator}lib$separator';
            final libIndex = absoluteImportPath.indexOf(libSearch);

            if (libIndex != -1) {
              final packageRelative = absoluteImportPath
                  .substring(libIndex + libSearch.length)
                  .replaceAll(separator, '/');
              final type = trimmed.startsWith('import') ? 'import' : 'export';
              final suffix = trimmed.substring(lastQuote + 1);
              final newLine =
                  "$type $quote"
                  "package:site_buddy/$packageRelative"
                  "$quote$suffix";

              changed = true;
              return line.replaceFirst(trimmed, newLine);
            }
          }
        }
      }
    }
    return line;
  }).toList();

  if (changed) {
    file.writeAsStringSync(newLines.join('\n'));
    return true;
  }
  return false;
}

String resolvePath(String base, String relative) {
  final baseUri = Uri.file(
    base.endsWith(Platform.pathSeparator)
        ? base
        : '$base${Platform.pathSeparator}',
  );
  return baseUri.resolve(relative).toFilePath();
}
